src-tauri/src/main.rs

```diff
@@ -17,13 +22,14 @@ fn main() {
 /// 
 /// list all directories in the base directory that have a `README.md` file inside
 #[tauri::command]
-async fn list() -> Vec<String> {
```

Comment:
@fersaru: note that I had to remove `async` due to some `tauri` issues: After adding the `config` argument (according to [tauri's documentation](https://tauri.app/v1/guides/features/command/), I had to choice between
- let the function return a `Result` or
- remove `async`

Reply:
I thought about a 'note' struct to handle the "true" backend and the main.rs becoming a presenter

Reply:
I'm also in favor of a `note` struct, but in this case we also need to inject it to tauri's commands (unless we introduce some singleton, which I don't want ;-)). By going back to async, we have to introduce a `Result`. I can do this.

Reply:
The only problem i would have when running a tauri command not async is when it causes the program to freeze.

Reply:
I made it async again.

> The only problem i would have when running a tauri command not async is when it causes the program to freeze.
I would not expect this to happen in our use case, since we don't intend to introduce heavy computation or long running blocking operations. _(Technically speaking, file system operations are blocking, but given a local filesystem, this should not result in a freeze.)_

Reply:
Also I'm still pondering about why a config would be a State.

Reply:
I think, the term `state` is poorly named by tauri. Looking at their own [example](https://tauri.app/v1/guides/features/command/#complete-example) they even use a database as "state". Other APIs I know use the term `context`, which suits the purpose betten in my opinion. However, once we have out `note` structure, we should use this as `state`. It should be sufficient to initialize our note using the config.